plugin/src/main/java/at/helpch/chatchat/data/impl/gson/GsonDatabase.java

```diff
@@ -0,0 +1,175 @@
+                plugin.getLogger().log(
+                    Level.WARNING,
+                    "Something went wrong while creating backup file. Shutting plugin down!",
+                    runtimeException
+                );
+
+                plugin.getServer().getPluginManager().disablePlugin(plugin);
+            }
+
+            final var user = new ChatUserImpl(uuid);
+            final var channel = plugin.configManager().channels().channels()
+                .get(plugin.configManager().channels().defaultChannel());
+
+            user.channel(channel);
+            return user;
+
+        } catch (final FileNotFoundException exception) { // Handles missing file
+            final var user = new ChatUserImpl(uuid);
+            final var channel = plugin.configManager().channels().channels()
+                .get(plugin.configManager().channels().defaultChannel());
+
+            user.channel(channel);
+            return user;
+        }
+    }
+
+    @Override
+    public void saveChatUser(@NotNull final ChatUser chatUser) {
+        final var userFile = new File(usersDirectory, chatUser.uuid() + ".json");
+
+        if (!userFile.exists()) {
+            try {
+                if (!userFile.createNewFile()) {
+                    plugin.getLogger().log(
+                        Level.WARNING,
+                        "Something went wrong while creating user file. Could not save data for user: " +
+                            chatUser.uuid()
+                    );
+                }
+            } catch (final IOException exception) {
+                plugin.getLogger().log(
+                    Level.WARNING,
+                    "Something went wrong while creating user file. Could not save data for user: " +
+                        chatUser.uuid(),
+                    exception
+                );
+            }
+        }
+
+        try (final var writer = new FileWriter(userFile)) {
+            gson.toJson(chatUser, writer);
+        } catch (final IOException exception) {
+            plugin.getLogger().log(
+                Level.WARNING,
+                "Something went wrong while saving user file. Could not save data for user: " +
+                    chatUser.uuid(),
+                exception
+            );
+        }
+    }
+
+    private static void copyFileContents(
+        @NotNull final File source,
+        @NotNull final File destination
+    ) throws RuntimeException {
+        try {
+            final var inStream = new FileInputStream(source);
+            final var outStream = new FileOutputStream(destination);
+
+            final byte[] buffer = new byte[1024];
+
+            int length;
+            while ((length = inStream.read(buffer)) > 0){
+                outStream.write(buffer, 0, length);
+            }
+
+            inStream.close();
+            outStream.close();
+        } catch (final IOException exception) {
+            throw new RuntimeException(exception);
```

Comment:
what is the point of this catch? seems useless.

Reply:
I just thought it would be better to have 1 single error thrown instead of multiple. I kinda forgot to add SecurityException there as well.